storage/storage-interface/src/state_view.rs

```diff
@@ -65,3 +71,33 @@ impl DbStateViewAtVersion for Arc<dyn DbReader> {
         })
     }
 }
+
+impl ModuleViewer for DbStateView {
+    type Error = anyhow::Error;
+    type Item = CompiledModule;
+
+    fn view_module(&self, module_id: &ModuleId) -> Result<Self::Item, Self::Error> {
+        let state_key = StateKey::access_path(AccessPath::from(module_id));
+        let state_value = self
+            .get_state_value(&state_key)?
+            .ok_or_else(|| anyhow!("Module {:?} not found", module_id))?;
+        CompiledModule::deserialize(state_value.bytes()).map_err(|e| {
+            anyhow!(
+                "Module {:?} failed to deserialize with error code {:?}",
+                module_id,
+                e,
+            )
+        })
+    }
+}
+
+// TODO: We can implement this for S: StateView but need to move the trait
```

Comment:
Oh, do we have the same issue for implementing ModuleViewer for StateView? What's the issue to already put it in aptos-types?

Reply:
We have a circular dependency in this case I believe. Let's ask Alden! @msmouse is there a reason why we have StateView not in `aptos-types`?

Reply:
Otherwise same issue, where the trait cannot be implemented for the arbitrary type outside of its definition, but moving the impl to definition creates the dependency.

Reply:
I don't quite remember :trollface: I'm fine moving it over.

Reply:
Well, let's change imports in 100+ files :trollface: